app/models/qa/linked_data/config/context_map.rb

```diff
@@ -0,0 +1,75 @@
+# Defines the external authority predicates used to extract additional context from the graph.
+module Qa
+  module LinkedData
+    module Config
+      class ContextMap
+        attr_reader :properties # [Array<Qa::LinkedData::Config::ContextPropertyMap>] set of property map models
+
+        attr_reader :context_map, :groups
+        private :context_map, :groups
+
+        # @param [Hash] context_map that defines groups and properties for additional context to display during the selection process
+        # @option context_map [Hash] :groups (optional) predefine group ids and labels to be used in the properties section to group properties
+        # @option groups [Hash] key=group_id; value=[Hash] with group_label_i18n and/or group_label_default
+        # @option context_map [Array<Hash>] :properties (optional) property maps defining how to get and display the additional context (if none, context will not be added)
+        # @example
+        #   {
+        #     "groups": {
+        #       "dates": {
+        #         "group_label_i18n": "qa.linked_data.authority.locnames_ld4l_cache.dates",
+        #         "group_label_default": "Dates",
+        #       }
+        #     },
+        #     "properties": [
+        #       {
+        #         "property_label_i18n": "qa.linked_data.authority.locgenres_ld4l_cache.authoritative_label",
+        #         "property_label_default": "Authoritative Label",
+        #         "lpath": "madsrdf:authoritativeLabel",
+        #         "selectable": true,
+        #         "drillable": false
+        #       },
+        #       {
+        #         "group_id": "dates",
+        #         "property_label_i18n": "qa.linked_data.authority.locnames_ld4l_cache.birth_date",
+        #         "property_label_default": "Birth",
+        #         "lpath": "madsrdf:identifiesRWO/madsrdf:birthDate/schema:label",
+        #         "selectable": false,
+        #         "drillable": false
+        #       }
+        #     ]
+        #   }
+        def initialize(context_map = {})
+          @context_map = context_map
+          extract_groups
+          extract_properties
+        end
+
+        def group_label(group_id)
```

Comment:
You are allowing the user to get a list of properties back, but not a list of groups.  They need to know the id to get at group information.  I'm guessing this is how you are using the data, but getting a list of groups seems like it might be nice.

Reply:
I am trying to only surface the public API what is actually being used.  When started out, I was surfacing what I thought would be useful and decided against that to avoid public API bloat.  If someone requests the list of groups, it can easily be added at that time.